lego/apps/events/serializers/events.py

```diff
@@ -284,6 +296,20 @@ def validate(self, data):
                         "end_time": "User does not have the required permissions for time travel"
                     }
                 )
+        if self.instance is not None and "share_info_flag" in data:
+            pools = self.instance.pools.all()
+            reg_time = min(pool.activation_date for pool in pools)
+            if data[
+                "share_info_flag"
+            ] != self.instance.share_info_flag and reg_time < datetime.now(
+                timezone.utc
+            ):
```

Comment:
You can do this a lot easier with a single query:
```python
self.instance.pools.filter(activation_time__lte=timezone.now()).exists()
```

Also, you can use `timezone.now()` instead of `datetime(...)` to get the correct timezone.